### PR TITLE
home-assistant-custom-components: add a bunch of stuff

### PR DESCRIPTION
**Pull request by K900**
## Description of changes

An update + some new components for stuff that I use, now that I'm finally migrating my HASS setup to nixpkgs components instead of HACS.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/midea-beautiful-air/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+, requests
+, pytestCheckHook
+, pytest-socket
+, requests-mock
+}:
+
+buildPythonPackage rec {
+  pname = "midea-beautiful-air";
+  version = "0.10.4";
+  format = "setuptools";
```
Comment (mweinelt) [inline pkgs/development/python-modules/midea-beautiful-air/default.nix:14]:
```suggestion
  pyproject = true;
```

And put `setuptools` into `build-system` instead.

File: pkgs/development/python-modules/midea-beautiful-air/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  src = fetchFromGitHub {
+    owner = "nbogojevic";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-1IOv9K8f69iRpYaCx3k0smVrCKPmDxlT/1uVoTyvIjU=";
+  };
+
+  propagatedBuildInputs = [
```
Comment (mweinelt) [inline pkgs/development/python-modules/midea-beautiful-air/default.nix:23]:
```suggestion
  dependencies = [
```

PEP517 lingo

File: pkgs/development/python-modules/midea-beautiful-air/default.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  ];
+
+  disabledTestPaths = [
+    # tests optional dependencies + network
+    "tests/test_cli.py"
+  ];
+
+  pythonImportsCheck = [ "midea_beautiful" ];
+
+  meta = with lib; {
```
Comment (mweinelt) [inline pkgs/development/python-modules/midea-beautiful-air/default.nix:41]: changelog https://github.com/nbogojevic/midea-beautiful-air/releases

File: pkgs/servers/home-assistant/custom-components/midea-air-appliances-lan/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  src = fetchFromGitHub {
+    inherit owner;
+    repo = "homeassistant-midea-air-appliances-lan";
+    rev = "v${version}";
+    hash = "sha256-CNvtpC7g0XR+cNDV9sMWc3sRLsy4bXFSDvkP+I7W1Gc=";
+  };
+
+  propagatedBuildInputs = [ midea-beautiful-air ];
+
+  meta = with lib; {
```
Comment (mweinelt) [inline pkgs/servers/home-assistant/custom-components/midea-air-appliances-lan/default.nix:21]: changelog https://github.com/nbogojevic/homeassistant-midea-air-appliances-lan/releases/tag/v0.9.2

File: pkgs/servers/home-assistant/custom-components/smartthinq-sensors/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    hash = "sha256-mt5/XHDAUeoMUA1jWdCNXTUgZBQkqabL5Y4MxwxcweY=";
+  };
+
+  propagatedBuildInputs = [
+    charset-normalizer
+    pycountry
+    xmltodict
+  ];
+
+  meta = with lib; {
```
Comment (mweinelt) [inline pkgs/servers/home-assistant/custom-components/smartthinq-sensors/default.nix:27]: changelog https://github.com/ollo69/ha-smartthinq-sensors/releases/tag/v0.39.0

File: pkgs/servers/home-assistant/custom-components/yassi/default.nix [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  src = fetchFromGitHub {
+    inherit owner;
+    repo = "ha_samsung_soundbar";
+    rev = version;
+    hash = "sha256-htAUCQe8mpk+GFwxXkPVnWS0m3mZd2hUt+f4qES+W4U=";
+  };
+
+  propagatedBuildInputs = [ pysmartthings ];
+
+  meta = with lib; {
```
Comment (mweinelt) [inline pkgs/servers/home-assistant/custom-components/yassi/default.nix:21]: changelog https://github.com/samuelspagl/ha_samsung_soundbar/releases/tag/0.3.0

**Review by mweinelt — CHANGES_REQUESTED** [review]

**Comment by K900** [comment]
Applied everything.

File: pkgs/development/python-modules/midea-beautiful-air/default.nix [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    description = "Python client for accessing Midea air conditioners and dehumidifiers (Midea, Comfee, Inventor EVO) via local network";
+    homepage = "https://github.com/nbogojevic/midea-beautiful-air";
+    changelog = "https://github.com/nbogojevic/midea-beautiful-air/releases/tag/v${version}";
+    maintainers = with maintainers; [ k900 ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/midea-beautiful-air/default.nix:50]:
```suggestion
    maintainers = with maintainers; [ k900 ];
    mainProgram = "midea-beautiful-air-cli";
```

Comment (K900) [reply pkgs/development/python-modules/midea-beautiful-air/default.nix:50]: Done

**Review by mweinelt — APPROVED** [review]

**Pull request merged**